src/llm_vm/completion/optimize.py

```diff
@@ -216,14 +215,29 @@ def complete_delay_train(self, stable_context, dynamic_prompt, c_id = None, **kw
             def promiseCompletion():
                 best_completion = self.call_big(prompt, **kwargs)
                 def actual_train(use_completion = None):
+                    def data_synthesis(prompt, response, **kwargs):
+                        final_prompt = '{"prompt": "' +prompt+'"  , "response": "' +response+'" }'+'\nGenerate 100 more JSONS each with a prompt and response field like the given one. The content of the prompt and response fields must be similar to the given JSON. Seperate each JSON with a ,.'
+                        data=self.call_big(final_prompt , kwargs)
+                        datapoints = []
+                        data = '{ "datapoints": ['+data+']}'
+                        print(data)
+                        data = json.loads(data)
+                        for i in data["datapoints"]:
+                            datapoints.append((i["prompt"],i["response"]))
+                        return datapoints
+
                     train_completion = best_completion if use_completion is None else use_completion
-                    new_datapoint = (dynamic_prompt.strip(), train_completion)
+                    new_datapoint = (prompt.strip(), train_completion)
```

Comment:
Why did you change dynamic_prompt to prompt?   It was originally dynamic_prompt so if the static_prompt was huge the smaller model wouldn't get it in the context window and it'd be even more optimized

Reply:
ok will change back, I probably got a bit confused by the way it was being used in test_optimize